### PR TITLE
revise benchmark dose modeling

### PR DESCRIPTION
**Pull request by shapiromatron**
Fix and modernize benchmark dose modeling. A complete rebuild of the BMD backend, and user-interface. All business logic for running BMD models and parsing inputs/outputs is done using the newly minted [BMDS](https://github.com/shapiromatron/bmds) package, which separates the business logic from the web-application. The frontend now uses React/Redux.

One major change in this approach is the BMD models are no longer run on the production webserver. In order to do this in the past, the BMDS model source code had to be recompiled for the Linux environment; it was only originally designed for Windows. This lead to large issues in version updates, and some small differences in the models themselves.

As an alternative, the BMDS module is monkeypatched to send requests to a remote (Windows) URL and outputs from these models are returned. This added new setting `BMD_HOST`, which is required for any BMD computation. 

The Windows implementation for running BMDS models is a lightweight PHP webserver, implemented by Liu Duan here: https://github.com/liuduan/hawc/tree/PC_Installation/Windows_server

### CODE REVIEW

**Comment by JoshAddington** [comment]
Should the dev setup documentation be revised to reflect any of these changes, or is it all handled by the pip install of bmds?

**Comment by shapiromatron** [comment]
Docs would probably be a good idea; the only setup change is documented in the `local.example.py` settings file.

**Comment by JoshAddington** [comment]
OK, cause I'm getting an exception running `python manage.py runserver`

```
Traceback (most recent call last):
  File "/Users/addingtonjd/.virtualenvs/hawc/lib/python2.7/site-packages/django/utils/autoreload.py", line 226, in wrapper
    fn(*args, **kwargs)
  File "/Users/addingtonjd/.virtualenvs/hawc/lib/python2.7/site-packages/django/core/management/commands/runserver.py", line 109, in inner_run
    autoreload.raise_last_exception()
  File "/Users/addingtonjd/.virtualenvs/hawc/lib/python2.7/site-packages/django/utils/autoreload.py", line 249, in raise_last_exception
    six.reraise(*_exception)
  File "/Users/addingtonjd/.virtualenvs/hawc/lib/python2.7/site-packages/django/utils/autoreload.py", line 226, in wrapper
    fn(*args, **kwargs)
  File "/Users/addingtonjd/.virtualenvs/hawc/lib/python2.7/site-packages/django/__init__.py", line 18, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/Users/addingtonjd/.virtualenvs/hawc/lib/python2.7/site-packages/django/apps/registry.py", line 108, in populate
    app_config.import_models(all_models)
  File "/Users/addingtonjd/.virtualenvs/hawc/lib/python2.7/site-packages/django/apps/config.py", line 202, in import_models
    self.models_module = import_module(models_module_name)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/Users/addingtonjd/Code/review/hawc/project/bmd/models.py", line 13, in <module>
    from .bmds_monkeypatch import bmds
  File "/Users/addingtonjd/Code/review/hawc/project/bmd/bmds_monkeypatch.py", line 63, in <module>
    bmds.Session.execute = execute
AttributeError: 'module' object has no attribute 'Session'
```

**Comment by shapiromatron** [comment]
I remember that issue. You'll need to delete all old pyc files. This should do the trick:

`rm -rf project/bmd/bmds`

**Comment by JoshAddington** [comment]
yep, that worked

File: project/bmd/models.py [hunk 1]
```diff
@@ -1,697 +1,362 @@
...
-        max_digits=5, decimal_places=2, null=True, blank=True,
-        help_text="If a threshold is required for the test, threshold can be specified to non-default.")
-    continuous_on = models.BooleanField(default=True, verbose_name="Continuous Datasets")
-    dichotomous_on = models.BooleanField(default=True, verbose_name="Dichotomous Datasets")
-    cancer_dichotomous_on = models.BooleanField(default=True, verbose_name="Cancer Dichotomous Datasets")
+class Model(models.Model):
 
-    class Meta:
-        ordering = ['logic_id']
+    IMAGE_UPLOAD_TO = 'bmds_plot'
```
Comment (JoshAddington) [inline project/bmd/models.py]: We should probably document that this is the name of the folder in MEDIA_ROOT where the bmds plots are being stored, and that the folder needs to be created.

Would it be better to have this as an env var?

Comment (shapiromatron) [inline project/bmd/models.py]: Well, it goes into the `MEDIA` path, which is standard django behavior. I'm surprised that this folder isn't created upon startup; maybe we could just add a check in `bmd/apps.py` in the startup to create this path if it doesn't exist?

Comment (JoshAddington) [inline project/bmd/models.py]: Yeah, that would clear up the IOErrors I was getting.

Comment (shapiromatron) [inline project/bmd/models.py]: Added new commit which should resolve this: https://github.com/shapiromatron/hawc/pull/27/commits/28b601829310caf1f5bc0d22dea887c5bac9992e

**Comment by JoshAddington** [comment]
Everything works for me. The only things I see is style stuff like 

```
import React, { Component, PropTypes } from 'react';

class DomainCell extends Component {}
```
vs. 
```
import React from 'react';

class DomainCell extends React.Component{}
```

kinda issues. If you want to merge and then decide on cohesive styling next week, that's fine with me.

**Comment by shapiromatron** [comment]
Sounds like a plan, thanks!

**Pull request merged**